src/vs/workbench/api/common/extHostTesting.ts

```diff
@@ -222,6 +228,110 @@ export class ExtHostTesting implements ExtHostTestingShape {
 		const { actual, previousChildren, previousEquals, ...item } = owned;
 		return Promise.resolve(item);
 	}
+
+	private createDefaultDocumentTestHierarchy(provider: vscode.TestProvider, document: vscode.TextDocument, folder: vscode.WorkspaceFolder | undefined): vscode.TestHierarchy<vscode.TestItem> | undefined {
+		if (!folder) {
+			return;
+		}
+
+		const workspaceHierarchy = provider.createWorkspaceTestHierarchy?.(folder);
+		if (!workspaceHierarchy) {
+			return;
+		}
+
+		const onDidChangeTest = new Emitter<vscode.TestItem>();
+		workspaceHierarchy.onDidChangeTest(node => {
+			const wrapper = TestItemFilteredWrapper.getWrapperForTestItem(node, document.uri);
+			if (wrapper.doesHaveNodeMatchingFilter(true)) {
+				onDidChangeTest.fire(wrapper);
+			} else {
+				onDidChangeTest.fire(wrapper.visibleParent);
```

Comment:
Related to the comment down below -- it's possible that the location will change for an existing testing. If this happens:

- If the before and after of `doesHaveNodeMatchingFilter` is the same, do nothing, otherwise;
- If the test was filtered out before, we should invalidated the cached `hasNodeMatchingFilter` up to the former `visibleParent` before firing the event (so the differ will see a new child that needs to be published);
- If the test existed before, do the same thing, but up to the _current_ visibleParent (so the subtree will be hidden)

Reply:
this should be addressed now